app/Models/GenereRepository.php

```
@@ -424,6 +462,75 @@ public function merge(int $sourceId, int $targetId): array
+            }
+            $positions = $srcPosStmt->get_result()->fetch_all(MYSQLI_ASSOC);
+
+            if (!empty($positions)) {
+                $findTargetPos = $this->db->prepare("SELECT id FROM posizioni WHERE scaffale_id = ? AND mensola_id = ? AND genere_id = ? LIMIT 1");
+                $moveBooksPos = $this->db->prepare("UPDATE libri SET posizione_id = ? WHERE posizione_id = ?");
+                $dropPos = $this->db->prepare("DELETE FROM posizioni WHERE id = ?");
+                $repointPos = $this->db->prepare("UPDATE posizioni SET genere_id = ? WHERE id = ?");
+                // Fail the transaction cleanly instead of fataling on bind_param
+                // if any statement could not be prepared.
+                if ($findTargetPos === false || $moveBooksPos === false || $dropPos === false || $repointPos === false) {
+                    throw new \RuntimeException('Errore nella preparazione delle query sulle posizioni');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '488', column '31'). (undefined)

(MissingImport)